### ISSUE TITLE
Some build environment variables are not passed in a custom plugin

### ISSUE THREAD

**Issue by metallica127** [argoproj/argo-cd#2971]
**Describe the bug**

According to [doc](https://argoproj.github.io/argo-cd/user-guide/config-management-plugins/#environment) custom plugins should have access to the standard build [environment variables](https://argoproj.github.io/argo-cd/user-guide/build-environment/). However, the following variables are missing:
- ARGOCD_APP_REVISION 
- ARGOCD_APP_SOURCE_PATH 
- ARGOCD_APP_SOURCE_REPO_URL
- ARGOCD_APP_SOURCE_TARGET_REVISION 

**To Reproduce**

Create a custom plugin following the [doc](https://argoproj.github.io/argo-cd/user-guide/config-management-plugins/) . In my case, it's just a simple go binary that is mounted in the repo-server pod. 
```yaml
apiVersion: v1
data:
  configManagementPlugins: |
    - name: env_test
      generate:
        command: ["env_test"]
        args: [""]  
```

**Expected behavior**

It is expected that all environment variables from the doc are available in a custom plugin.

**Version**

```bash
$ argocd version
argocd: v1.3.2+a5a65cd
  BuildDate: 2019-12-03T21:31:16Z
  GitCommit: a5a65cdfe7c43fd1e3beada202d80c19bc50bd40
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:47:48Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: Version: {Version:kustomize/v3.2.1 GitCommit:d89b448c745937f0cf1936162f26a5aac688f840 BuildDate:2019-09-27T00:10:52Z GoOs:linux GoArch:amd64}
  Helm Version: v2.15.2
  Kubectl Version: v1.14.0
```

**Logs**
The actual env that is dumped by plugin:
```bash
KUBERNETES_SERVICE_PORT_HTTPS=443 
ARGOCD_DEX_SERVER_SERVICE_PORT=5556 
ARGOCD_DEX_SERVER_PORT_5557_TCP_PORT=5557 
ARGOCD_SERVER_SERVICE_PORT_HTTPS=443 
KUBERNETES_SERVICE_PORT=443 
ARGOCD_SERVER_PORT_80_TCP_PORT=80 
ARGOCD_SERVER_PORT_80_TCP_ADDR=172.30.161.97 
KUBERNETES_PORT_53_TCP_PORT=53 
ARGOCD_METRICS_PORT_8082_TCP_ADDR=172.30.95.45 
HOSTNAME=argocd-repo-server-65575964b-5nc79 
ARGOCD_METRICS_SERVICE_HOST=172.30.95.45 
ARGOCD_DEX_SERVER_SERVICE_PORT_GRPC=5557 
ARGOCD_SERVER_METRICS_PORT=tcp://172.30.176.46:8083 
ARGOCD_REPO_SERVER_PORT_8081_TCP=tcp://172.30.195.126:8081 
KUBERNETES_PORT_53_UDP_PORT=53 
ARGOCD_REDIS_SERVICE_PORT=6379 
ARGOCD_METRICS_PORT=tcp://172.30.95.45:8082 
ARGOCD_REDIS_SERVICE_PORT_TCP_REDIS=6379 
ARGOCD_REPO_SERVER_PORT_8081_TCP_ADDR=172.30.195.126 
PWD=/home/argocd 
ARGOCD_REDIS_PORT_6379_TCP_ADDR=172.30.91.139 
ARGOCD_METRICS_PORT_8082_TCP_PROTO=tcp 
ARGOCD_REDIS_PORT_6379_TCP=tcp://172.30.91.139:6379 
ARGOCD_SERVER_SERVICE_PORT_HTTP=80 
ARGOCD_SERVER_METRICS_PORT_8083_TCP_PROTO=tcp 
ARGOCD_SERVER_PORT_80_TCP=tcp://172.30.161.97:80 
ARGOCD_DEX_SERVER_PORT_5556_TCP=tcp://172.30.158.99:5556 
KUBERNETES_PORT_53_TCP=tcp://172.30.0.1:53 
ARGOCD_METRICS_PORT_8082_TCP=tcp://172.30.95.45:8082 
KUBERNETES_PORT_53_UDP=udp://172.30.0.1:53 
HOME=/home/argocd 
KUBERNETES_PORT_443_TCP=tcp://172.30.0.1:443 
ARGOCD_SERVER_METRICS_PORT_8083_TCP_PORT=8083 
ARGOCD_REPO_SERVER_PORT_8084_TCP_PROTO=tcp 
ARGOCD_DEX_SERVER_PORT_5556_TCP_PROTO=tcp 
ARGOCD_REDIS_SERVICE_HOST=172.30.91.139 
KUBERNETES_SERVICE_PORT_DNS=53 
ARGOCD_REPO_SERVER_PORT_8084_TCP_PORT=8084 
ARGOCD_REPO_SERVER_PORT=tcp://172.30.195.126:8081 
ARGOCD_REDIS_PORT_6379_TCP_PORT=6379 
ARGOCD_REPO_SERVER_SERVICE_PORT_METRICS=8084 
ARGOCD_SERVER_METRICS_SERVICE_PORT=8083 
ARGOCD_SERVER_PORT_443_TCP=tcp://172.30.161.97:443 
ARGOCD_SERVER_PORT_80_TCP_PROTO=tcp 
ARGOCD_SERVER_SERVICE_HOST=172.30.161.97 
KUBERNETES_PORT_53_TCP_PROTO=tcp 
ARGOCD_REPO_SERVER_SERVICE_HOST=172.30.195.126 
ARGOCD_REPO_SERVER_PORT_8084_TCP=tcp://172.30.195.126:8084 
ARGOCD_DEX_SERVER_PORT_5557_TCP_ADDR=172.30.158.99 
KUBERNETES_SERVICE_PORT_DNS_TCP=53 
USER=argocd 
ARGOCD_SERVER_SERVICE_PORT=80 
ARGOCD_SERVER_PORT_443_TCP_PORT=443 
ARGOCD_REPO_SERVER_PORT_8084_TCP_ADDR=172.30.195.126 
ARGOCD_METRICS_PORT_8082_TCP_PORT=8082 
ARGOCD_SERVER_METRICS_SERVICE_PORT_METRICS=8083 
SHLVL=0 
ARGOCD_DEX_SERVER_PORT_5557_TCP_PROTO=tcp 
KUBERNETES_PORT_443_TCP_PROTO=tcp 
ARGOCD_DEX_SERVER_PORT_5557_TCP=tcp://172.30.158.99:5557 
KUBERNETES_PORT_443_TCP_ADDR=172.30.0.1 
ARGOCD_REDIS_PORT=tcp://172.30.91.139:6379 
KUBERNETES_PORT_53_UDP_ADDR=172.30.0.1 
ARGOCD_METRICS_SERVICE_PORT=8082 
ARGOCD_SERVER_METRICS_PORT_8083_TCP_ADDR=172.30.176.46 
ARGOCD_SERVER_METRICS_PORT_8083_TCP=tcp://172.30.176.46:8083 
ARGOCD_REPO_SERVER_SERVICE_PORT=8081 
KUBERNETES_SERVICE_HOST=172.30.0.1 
KUBERNETES_PORT_53_TCP_ADDR=172.30.0.1 
KUBERNETES_PORT=tcp://172.30.0.1:443 
KUBERNETES_PORT_443_TCP_PORT=443 
KUBERNETES_PORT_53_UDP_PROTO=udp 
ARGOCD_SERVER_METRICS_SERVICE_HOST=172.30.176.46 
ARGOCD_DEX_SERVER_PORT_5556_TCP_PORT=5556 
ARGOCD_REPO_SERVER_SERVICE_PORT_SERVER=8081 
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin 
ARGOCD_DEX_SERVER_PORT_5556_TCP_ADDR=172.30.158.99 
ARGOCD_SERVER_PORT_443_TCP_ADDR=172.30.161.97 
ARGOCD_DEX_SERVER_SERVICE_HOST=172.30.158.99 
ARGOCD_METRICS_SERVICE_PORT_METRICS=8082 
ARGOCD_REPO_SERVER_PORT_8081_TCP_PORT=8081 
ARGOCD_DEX_SERVER_SERVICE_PORT_HTTP=5556 
ARGOCD_REDIS_PORT_6379_TCP_PROTO=tcp 
ARGOCD_REPO_SERVER_PORT_8081_TCP_PROTO=tcp 
ARGOCD_SERVER_PORT_443_TCP_PROTO=tcp 
ARGOCD_SERVER_PORT=tcp://172.30.161.97:80 
ARGOCD_DEX_SERVER_PORT=tcp://172.30.158.99:5556 
ARGOCD_APP_NAME=myapp
ARGOCD_APP_NAMESPACE=myapp
GIT_SSH_COMMAND=ssh -i /dev/shm/289433057 -o StrictHostKeyChecking=yes -o UserKnownHostsFile=/app/config/ssh/ssh_known_hosts 
```

**Comment by alexmt** [comment]
hello @metallica127 , 

The env variables are available in v1.4 release. The 1.4 release candidate was published this morning: https://github.com/argoproj/argo-cd/releases/tag/v1.4.0-rc1

**Comment by alexmt** [comment]
Closing the ticket. @metallica127, please let me know if you think it is too early to close the ticket.
